collect_app/src/test/java/org/odk/collect/android/configure/qr/QRCodeActivityResultDelegateTest.java

```diff
@@ -53,62 +57,72 @@ public QRCodeDecoder providesQRCodeDecoder() {
     }
 
     @Test
-    public void forSelectPhoto_importsSettingsFromQRCode_showsSuccessToast() {
-        QRCodeActivityResultDelegate delegate = new QRCodeActivityResultDelegate(context, settingsImporter, fakeQRDecoder);
+    public void forSelectPhoto_importsSettingsFromQRCode_showsSuccessToast_andLogsAnalytics() {
```

Comment:
Yay for delegate tests! I think here I'd personally split the tests so that each "then" of a "given"/"when" gets its own test. Can just make it easier to navigate the assertions and change individual features. The trade off is run time of course but these tests are pretty fast once Robolectric has booted up (which is a small cost when you're running everything).